compiler/noirc_evaluator/src/ssa/opt/constant_folding.rs

```diff
@@ -317,531 +317,305 @@ mod test {
 
     use crate::ssa::{
         function_builder::FunctionBuilder,
-        ir::{
-            instruction::{Binary, BinaryOp, Instruction, TerminatorInstruction},
-            map::Id,
-            types::Type,
-            value::{Value, ValueId},
-        },
+        ir::{map::Id, types::Type},
+        opt::assert_ssa_equals,
+        Ssa,
     };
-    use acvm::{acir::AcirField, FieldElement};
 
     #[test]
     fn simple_constant_fold() {
-        // fn main f0 {
-        //   b0(v0: Field):
-        //     v1 = add v0, Field 1
-        //     v2 = mul v1, Field 3
-        //     return v2
-        // }
-        //
         // After constructing this IR, we set the value of v0 to 2.
         // The expected return afterwards should be 9.
-        let main_id = Id::test_new(0);
-
-        // Compiling main
-        let mut builder = FunctionBuilder::new("main".into(), main_id);
-        let v0 = builder.add_parameter(Type::field());
-
-        let one = builder.field_constant(1u128);
-        let two = builder.field_constant(2u128);
-        let three = builder.field_constant(3u128);
-
-        let v1 = builder.insert_binary(v0, BinaryOp::Add, one);
-        let v2 = builder.insert_binary(v1, BinaryOp::Mul, three);
-        builder.terminate_with_return(vec![v2]);
-
-        let mut ssa = builder.finish();
+        let src = "
+            acir(inline) fn main f0 {
+              b0(v0: Field):
+                v1 = add v0, Field 1
+                v2 = mul v1, Field 3
+                return v2
+            }
+            ";
+        let mut ssa = Ssa::from_str(src).unwrap();
```

Comment:
Right, I thought about this but was too lazy to do it. I'd need to translate spans to lines/columns... the error is more r less understandable, like it would say that it found some token when it was expecting some other token, so you could guess where the error is. But I agree that showing the error where it's located will be better for all of us in the future... but I think `unwrap` will still show the `Debug` version of the error? Or how does it work?

Reply:
Yes I think `.expect` and `.unwrap` need `Debug`. Just kinda remember the raw compilation errors being somewhat difficult to parse in test failures, but we can add a helper any time 👍